File: db_templates/Java-JUnit/Root/test/FinderTest.java
/**
 * Tests for class Finder.
 * 
 * All tests in the folder "test" are executed 
 * when the "Test" action is invoked.
 * 
 */

import static org.junit.Assert.*;
import org.junit.Test;


public class FinderTest {

    @Test
    public final void testFindMaximumElement1() {

        // we define some test input and what result we would expect
        int[] testInput = new int[] { 1, 2, 3, 4, 5 };
        int expectedResult = 5;

        // create a Finder object and call the findMaximumElement function
        // with test input
        Finder myFinder = new Finder();
        int actualResult = myFinder.findMaximumElement(testInput);

        // the actualResult value should be the same as the expectedResult value
        assertTrue("Test input with ascending order",
                actualResult == expectedResult);
    }

    @Test
    public final void testFindMaximumElement2() {

        int[] testInput = new int[] { 117, 56, 38, 11, 0 };
        int expectedResult = 117;

        Finder myFinder = new Finder();
        int actualResult = myFinder.findMaximumElement(testInput);

        assertTrue("Test input with descending order",
                actualResult == expectedResult);
    }

    @Test
    public final void testFindMaximumElement3() {

        int[] testInput = new int[] { 42, 11, 38, 75, 14 };
        int expectedResult = 75;

        Finder myFinder = new Finder();
        int actualResult = myFinder.findMaximumElement(testInput);

        assertTrue("Test input with random order",
                actualResult == expectedResult);
    }

}

File: distRepo/lib/config/db_templates/Python-UnitTest/Root/test/finderTest.py
##
# Tests for finder.py
#
# All tests in the folder "test" are executed
# when the "Test" action is invoked.
#
##

from Root.src.finder import maximum_in_list
import unittest

class finderTest(unittest.TestCase):

    # test with a list sorted in ascending order
    def test_ascending_input(self):
        self.assertEqual(maximum_in_list([1, 2, 3, 4, 5]), 5)

    # test with a list sorted in descending order
    def test_descending_input(self):
        self.assertEqual(maximum_in_list([117, 56, 38, 11, 0]), 117)

    # test with a randomly sorted list
    def test_random_input(self):
        self.assertEqual(maximum_in_list([42, 11, 38, 75, 14]), 75)


if __name__ == '__main__':
    unittest.main()
